core/node/via_btc_watch/src/lib.rs

```diff
@@ -1,20 +1,26 @@
+mod message_processors;
+
 use std::time::Duration;
 
 use anyhow::Context as _;
 use tokio::sync::watch;
-use via_btc_client::{indexer::BitcoinInscriptionIndexer, types::BitcoinTxid};
+use via_btc_client::{
+    indexer::BitcoinInscriptionIndexer,
+    types::{BitcoinAddress, BitcoinNetwork, BitcoinTxid},
+};
 use zksync_dal::{Connection, ConnectionPool, Core, CoreDal};
+use zksync_types::PriorityOpId;
 
-use crate::message_processors::MessageProcessorError;
-
-mod message_processors;
+use self::message_processors::{L1ToL2MessageProcessor, MessageProcessor, MessageProcessorError};
 
-// re-exporting here isn't necessary, but it's a good practice to keep all the public types in one place
-pub use via_btc_client::types::BitcoinNetwork;
+// Number of blocks that we should wait before processing the new blocks.
+pub(crate) const BTC_BLOCKS_LAG: u32 = 1000;
 
 #[derive(Debug)]
 struct BtcWatchState {
     last_processed_bitcoin_block: u32,
+    next_expected_priority_id: PriorityOpId,
```

Comment:
i saw you created the next_expected_priority_id based on transaction serial id and vector length


could you please explain a little bit about the calculation and storing of the  `PriorityOpId`?

it's dependent on memory or you reading some data from db?
what is serial_id and how it's related to `PriorityOpId`

Reply:
basically it's the same thing. unique id for each transaction (check the transaction dal for more info)